hack/quickstart-setup.sh

```diff
@@ -0,0 +1,322 @@
+  echo "Enter an AWS secret access key id for an Account where you have access to Route53:"
+  read MGC_AWS_ACCESS_KEY_ID </dev/tty
+  echo "export MGC_AWS_ACCESS_KEY_ID for future executions of the script to skip this step"
+fi
+
+if [[ -z "${MGC_AWS_SECRET_ACCESS_KEY}" ]]; then
+  echo "Enter your AWS secret access key id for an Account where you have access to Route53:"
+  read MGC_AWS_SECRET_ACCESS_KEY </dev/tty
+  echo "export MGC_AWS_SECRET_ACCESS_KEY for future executions of the script to skip this step"
+fi
+
+if [[ -z "${MGC_AWS_REGION}" ]]; then
+  echo "Enter an AWS region (e.g. eu-west-1) for an Account where you have access to Route53:"
+  read MGC_AWS_REGION </dev/tty
+  echo "export MGC_AWS_REGION for future executions of the script to skip this step"
+fi
+
+if [[ -z "${MGC_AWS_DNS_PUBLIC_ZONE_ID}" ]]; then
+  echo "Enter the Public Zone ID of your Route53 zone:"
+  read MGC_AWS_DNS_PUBLIC_ZONE_ID </dev/tty
+  echo "export MGC_AWS_DNS_PUBLIC_ZONE_ID for future executions of the script to skip this step"
+fi
+
+if [[ -z "${MGC_ZONE_ROOT_DOMAIN}" ]]; then
+  echo "Enter the root domain of your Route53 hosted zone (e.g. www.example.com):"
+  read MGC_ZONE_ROOT_DOMAIN </dev/tty
+  echo "export MGC_ZONE_ROOT_DOMAIN for future executions of the script to skip this step"
+fi
+
+# Default config
+if [[ -z "${LOG_LEVEL}" ]]; then
+  LOG_LEVEL=1
+fi
+if [[ -z "${OCM_SINGLE}" ]]; then
+  OCM_SINGLE=true
+fi
+if [[ -z "${MGC_WORKLOAD_CLUSTERS_COUNT}" ]]; then
+  MGC_WORKLOAD_CLUSTERS_COUNT=1
+fi
+
+cleanup
+
+port80=9090
+port443=8445
+proxyPort=9200
+metalLBSubnetStart=200
+
+# Create network for the clusters
+docker network create -d bridge --subnet 172.32.0.0/16 mgc --gateway 172.32.0.1 \
+  -o "com.docker.network.bridge.enable_ip_masquerade"="true" \
+  -o "com.docker.network.driver.mtu"="1500"
+
+# Create Kind control plane cluster
+kindCreateCluster ${KIND_CLUSTER_CONTROL_PLANE} ${port80} ${port443}
+
+# Deploy OCM hub
+deployOCMHub ${KIND_CLUSTER_CONTROL_PLANE}
+
+# Deploy Quick start kustomize
+deployQuickStartHub ${KIND_CLUSTER_CONTROL_PLANE}
+
+# Deploy cert manager
+configureCertManager ${KIND_CLUSTER_CONTROL_PLANE}
+
+# Initialize local dev setup for the controller on the control-plane cluster
+configureController ${KIND_CLUSTER_CONTROL_PLANE}
+
+# Deploy MetalLb
+configureMetalLB ${KIND_CLUSTER_CONTROL_PLANE} ${metalLBSubnetStart}
+
+configureControlCluster ${KIND_CLUSTER_CONTROL_PLANE}
+
+# Add workload clusters if MGC_WORKLOAD_CLUSTERS_COUNT environment variable is set
+if [[ -n "${MGC_WORKLOAD_CLUSTERS_COUNT}" ]]; then
+  for ((i = 1; i <= ${MGC_WORKLOAD_CLUSTERS_COUNT}; i++)); do
+    kindCreateCluster ${KIND_CLUSTER_WORKLOAD}-${i} $((${port80} + ${i})) $((${port443} + ${i})) $((${i} + 1))
+    deployIstio ${KIND_CLUSTER_WORKLOAD}-${i}
+    deployQuickStartSpoke ${KIND_CLUSTER_WORKLOAD}-${i}
+    deployOLM ${KIND_CLUSTER_WORKLOAD}-${i}
+    deployOCMSpoke ${KIND_CLUSTER_WORKLOAD}-${i}
+  done
+fi
```

Comment:
You might want to consider disentangling the creation of the clusters from the provisioning of components on them, this was something i was hoping would be done for local-setup as part of https://github.com/Kuadrant/multicluster-gateway-controller/issues/256, but for some reason has been stopped.

Doing it this way might not be a huge issue for you in a quick start, if it fails i guess you will just have to start again anyway, but it's really frustrating when it fails to provision a cluster after already creating and provisioning others, and the only option is to start completely from scratch.

You could iterate and create the kind clusters first see https://github.com/mikenairn/multi-cluster-traffic-controller/blob/56059eb0fe82fcf99247aa2fffcd68f07aef331d/hack/local-setup.sh#L231-L243

And then iterate again to provision each with mgc components see https://github.com/mikenairn/multi-cluster-traffic-controller/blob/56059eb0fe82fcf99247aa2fffcd68f07aef331d/hack/local-setup.sh#L247-L260

This isn't something i would expect to block a merge, but i think separating those two things would help provide a better flow in the quick start 

1. Create my infra (Kind Clusters). This part could for example eventually be updated to have you select existing clusters (Clusters already running on public clouds)
2. Provision stuff on created/selected clusters

Reply:
@mikenairn i've incorporated some of the changes suggested here and pushed them in the latest pitstop commit. I agree it's a lot cleaner and was able to remove more of the scripting. There's probably some further cleanup that could be done. I refactored the control/workload kustomizes to match what you had. It looks cleaner. 👍